orttraining/orttraining/test/python/checkpoint/orttraining_test_load_checkpoint.py

```diff
@@ -0,0 +1,1384 @@
+#!/usr/bin/env python3
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import os
+import pickle
+from numpy.testing import assert_allclose
+import numpy as np
+import argparse
+import glob
+
+import torch
+import torch.distributed as dist
+
+import onnxruntime
+from onnxruntime.training import amp, checkpoint, optim, orttrainer
+from _test_helpers import distributed_setup, load_evaluate_extract_state_dict, split_state_dict, split_name, aggregate_states
+
+def test_load_from_single_node_full_precision_into_single_node_full_precision(device = 'cuda', checkpoint_dir = 'checkpoint_dir/single_node/full_precision/'):
+    opts = {'device' : {'id' : device},
+            'debug' : {'deterministic_compute': True}}
+    
+    # extract state dictionaries to compare
+    state_dict_post_checkpoint, model = load_evaluate_extract_state_dict(device, orttrainer.ORTTrainerOptions(opts), checkpoint_dir)
+
+    state = pickle.load(open(os.path.join(checkpoint_dir, 'state_dict.pkl'), 'rb'))
```

Comment:
Dont you have to call `close()` on the file handle to prevent memory leak?
```suggestion
    with open(os.path.join(checkpoint_dir, 'state_dict.pkl'), 'rb') as f:
        state = pickle.load(f)
```

Reply:
yeah, I need to use with open. I forgot to change that. Updating.